url73e3057d61_empeeoh_BACnet4J_tgz-pJ8-com_serotonin_bacnet4j_test_OverflowTestJ8/src/org/free/bacnet4j/util/IpAddressUtils.java

```diff
@@ -0,0 +1,138 @@
+													throws IpWhiteListException{
+		if("*".equals(allowed))
+			return true;
+		int dash = allowed.indexOf('-');
+		int from;
+		int to;
+		int rem;
+		try{
+			if(dash == -1)
+				return Integer.parseInt(allowed) == Integer.parseInt(remote);
+		} catch(NumberFormatException e){
+			throw new IpWhiteListException((new StringBuilder())
+									  .append("Integer parsing error. allowed=")
+									  .append(allowedIp).append(", remote=")
+									  .append(remoteIp).toString());
+		}
+		from = Integer.parseInt(allowed.substring(0, dash));
+		to = Integer.parseInt(allowed.substring(dash + 1));
+		rem = Integer.parseInt(remote);
+		return from <= rem && rem <= to;
+	}
+
+	public static String checkIpMask(String ip){
+		String ipParts[] = ip.split("\\.");
+		if(ipParts.length != 4)
+			return "IP address must have 4 parts";
+		String message = checkIpMaskPart(ipParts[0]);
+		if(message != null)
+			return message;
+		message = checkIpMaskPart(ipParts[1]);
+		if(message != null)
+			return message;
+		message = checkIpMaskPart(ipParts[2]);
+		if(message != null)
+			return message;
+		message = checkIpMaskPart(ipParts[3]);
+		if(message != null)
+			return message;
+		else
+			return null;
+	}
+
+	private static String checkIpMaskPart(String part){
+		int dash;
+		if("*".equals(part))
+			return null;
+		dash = part.indexOf('-');
+		if(dash == -1){
+			int value = Integer.parseInt(part);
+			if(value < 0 || value > 255)
+				return (new StringBuilder()).append("Value out of range in '")
+										   .append(part).append("'").toString();
+			//break MISSING_BLOCK_LABEL_219;
+		}
+		int from;
+		from = Integer.parseInt(part.substring(0, dash));
+		if(from < 0 || from > 255)
+			return (new StringBuilder()).append("'From' value out of range in '").append(part).append("'").toString();
+		int to;
+		try{
+			to = Integer.parseInt(part.substring(dash + 1));
+			if(to < 0 || to > 255)
+				return (new StringBuilder()).append("'To' value out of range in '").append(part).append("'").toString();
+		}
+		catch(NumberFormatException e){
+			return (new StringBuilder()).append("Integer parsing error in '").append(part).append("'").toString();
+		}
+		if(from > to)
+			return (new StringBuilder()).append("'From' value is greater than 'To' value in '").append(part).append("'").toString();
+		return null;
+	}
+
+	public static byte[] toIpAddress(String addr) throws IllegalArgumentException{
+		if(addr == null)
+			throw new IllegalArgumentException("Invalid address: (null)");
+		String parts[] = addr.split("\\.");
+		if(parts.length != 4)
+			throw new IllegalArgumentException("IP address must have 4 parts");
+		byte ip[] = new byte[4];
+		for(int i = 0; i < 4; i++)
+			try{
+				int part = Integer.parseInt(parts[i]);
+				if(part < 0 || part > 255)
+					throw new IllegalArgumentException((new StringBuilder()).append("Value out of range in '").append(parts[i]).append("'").toString());
+				ip[i] = (byte)part;
+		}
+		catch(NumberFormatException e){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)